src/inbox/mod_inbox_muc.erl

```diff
@@ -0,0 +1,94 @@
+%%%-------------------------------------------------------------------
+%%% @author Dominik Stanaszek dominik.stanaszek@erlang-solutions.com
+%%% @copyright (C) 2018, Erlang-Solutions
+%%% @doc
+%%%
+%%% @end
+%%% Created : 6.07.2018 
+%%%-------------------------------------------------------------------
+-module(mod_inbox_muc).
+-author("dominik.stanaszek@erlang-solutions.com").
+-include("jlib.hrl").
+
+-export([update_inbox/5, start/1, stop/1]).
+
+start(Host) ->
+    ejabberd_hooks:add(update_inbox, Host, ?MODULE, update_inbox, 90),
+    % TODO check ooptions: if system messages stored -> 
+    % add hook handler for system messages on hook ie. invitation_sent
+    ok.
+
+stop(Host) ->
+    ejabberd_hooks:delete(update_inbox, Host, ?MODULE, update_inbox, 90),
+    ok.
+
+update_inbox(Acc, Room, {From, FromRoomJid}, AffsDict, Packet) ->
+    Affs = dict:to_list(AffsDict),
```

Comment:
Generally, dict:fold/3 should be used in such cases. BUT don't touch this code NOW, because we would replace dicts with maps pretty soon and we would touch the code.